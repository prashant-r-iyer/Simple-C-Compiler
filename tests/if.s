	.text

.globl main
main:
	# Save Frame pointer
	pushq %rbp
	movq %rsp, %rbp
	subq $256, %rsp
	pushq %rbx
	pushq %rbx
	pushq %r10
	pushq %r13
	pushq %r14
	pushq %r15
	pushq %r11
	pushq %r9

	# push 1
	movq $1, %rbx
	cmpq $0, %rbx
	je else_1
	movq $string0, %rbx
	# func=printf nargs=1

	# Move values from reg stack to reg args
	movq %rbx, %rdi
	movl $0, %eax
	call printf
	movq %rax, %rbx
	jmp if_end_1

else_1:

if_end_1:

	# push 0
	movq $0, %rbx
	cmpq $0, %rbx
	je else_2
	movq $string1, %rbx
	# func=printf nargs=1

	# Move values from reg stack to reg args
	movq %rbx, %rdi
	movl $0, %eax
	call printf
	movq %rax, %rbx
	jmp if_end_2

else_2:

if_end_2:

	# push 1
	movq $1, %rbx
	cmpq $0, %rbx
	je else_3
	movq $string2, %rbx
	# func=printf nargs=1

	# Move values from reg stack to reg args
	movq %rbx, %rdi
	movl $0, %eax
	call printf
	movq %rax, %rbx
	jmp if_end_3

else_3:
	movq $string3, %rbx
	# func=printf nargs=1

	# Move values from reg stack to reg args
	movq %rbx, %rdi
	movl $0, %eax
	call printf
	movq %rax, %rbx

if_end_3:

	# push 0
	movq $0, %rbx
	cmpq $0, %rbx
	je else_4
	movq $string4, %rbx
	# func=printf nargs=1

	# Move values from reg stack to reg args
	movq %rbx, %rdi
	movl $0, %eax
	call printf
	movq %rax, %rbx
	jmp if_end_4

else_4:
	movq $string5, %rbx
	# func=printf nargs=1

	# Move values from reg stack to reg args
	movq %rbx, %rdi
	movl $0, %eax
	call printf
	movq %rax, %rbx

if_end_4:
	movq $string6, %rbx
	# func=printf nargs=1

	# Move values from reg stack to reg args
	movq %rbx, %rdi
	movl $0, %eax
	call printf
	movq %rax, %rbx

	# Restore registers
	popq %r9
	popq %r11
	popq %r15
	popq %r14
	popq %r13
	popq %r10
	popq %rbx
	popq %rbx
	addq $256, %rsp

	leave
	ret

string0:
	.string "OK1\n"

string1:
	.string "OK2\n"

string2:
	.string "OK3\n"

string3:
	.string "OK4\n"

string4:
	.string "OK5\n"

string5:
	.string "OK6\n"

string6:
	.string "OK7\n"

