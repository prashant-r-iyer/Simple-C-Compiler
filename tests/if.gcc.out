if.c: In function ‘main’:
if.c:5:17: warning: implicit declaration of function ‘printf’ [-Wimplicit-function-declaration]
    5 |                 printf("OK1\n");
      |                 ^~~~~~
if.c:1:1: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
  +++ |+#include <stdio.h>
    1 | 
if.c:5:17: warning: incompatible implicit declaration of built-in function ‘printf’ [-Wbuiltin-declaration-mismatch]
    5 |                 printf("OK1\n");
      |                 ^~~~~~
if.c:5:17: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
if.c:9:17: warning: incompatible implicit declaration of built-in function ‘printf’ [-Wbuiltin-declaration-mismatch]
    9 |                 printf("OK2\n");
      |                 ^~~~~~
if.c:9:17: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
if.c:13:17: warning: incompatible implicit declaration of built-in function ‘printf’ [-Wbuiltin-declaration-mismatch]
   13 |                 printf("OK3\n");
      |                 ^~~~~~
if.c:13:17: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
if.c:16:17: warning: incompatible implicit declaration of built-in function ‘printf’ [-Wbuiltin-declaration-mismatch]
   16 |                 printf("OK4\n");
      |                 ^~~~~~
if.c:16:17: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
if.c:20:17: warning: incompatible implicit declaration of built-in function ‘printf’ [-Wbuiltin-declaration-mismatch]
   20 |                 printf("OK5\n");
      |                 ^~~~~~
if.c:20:17: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
if.c:23:17: warning: incompatible implicit declaration of built-in function ‘printf’ [-Wbuiltin-declaration-mismatch]
   23 |                 printf("OK6\n");
      |                 ^~~~~~
if.c:23:17: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
if.c:26:9: warning: incompatible implicit declaration of built-in function ‘printf’ [-Wbuiltin-declaration-mismatch]
   26 |         printf("OK7\n");
      |         ^~~~~~
if.c:26:9: note: include ‘<stdio.h>’ or provide a declaration of ‘printf’
